apollo-api/src/jvmMain/kotlin/com/apollographql/apollo3/api/-Data.kt

```diff
@@ -0,0 +1,27 @@
+package com.apollographql.apollo3.api
+
+import com.apollographql.apollo3.api.json.JsonWriter
+
+private fun Operation.Data.adapter(): Adapter<Operation.Data> {
+  val name = this::class.java.name
+
+  val operationQualifiedName = name.removeSuffix("${'$'}Data")
+  val operationName = operationQualifiedName.substringAfterLast(".")
+  val packageName = operationQualifiedName.substringBeforeLast(".")
+
+  val adapterName = "$packageName.adapter.${operationName}_ResponseAdapter${'$'}Data"
+
+  val clazz = Class.forName(adapterName)
+
+  val field = clazz.getDeclaredField("INSTANCE")
+
+  @Suppress("UNCHECKED_CAST")
+  val adapter = field.get(null) as Adapter<Operation.Data>
+
+  return adapter.obj()
+}
+
+fun Operation.Data.toJson(jsonWriter: JsonWriter, customScalarAdapters: CustomScalarAdapters = CustomScalarAdapters.Empty) {
```

Comment:
Should we mark it `@ApolloExperimental`?

Reply:
I was wondering the same. I decided not to because it's 2.x functionality for which we don't have a multiplatform equivalent yet. So it's not really new and it's not really deprecated either. 

Once we have a multiplatform mockserver API, we'll mark this one as deprecated?

Reply:
Sounds good 👍